File: backend/xchange_api/src/main/java/br/com/xchange/api/application/dto/response/CoinDetailResponse.java
package br.com.xchange.api.application.dto.response;

import java.math.BigDecimal;
import java.util.List;

public record CoinDetailResponse(
    String id,
    String name,
    String symbol,
    String imageUrl,
    String description,
    String hashingAlgorithm,
    String genesisDate,
    Integer marketCapRank,
    Long watchlistPortfolioUsers,
    BigDecimal sentimentVotesUpPercentage,
    BigDecimal sentimentVotesDownPercentage,

    String price,
    String high24h,
    String low24h,
    String priceChange24h,
    String marketCap,
    String totalVolume,
    String fullyDilutedValuation,

    BigDecimal priceChangePercentage1h,
    BigDecimal priceChangePercentage24h,
    BigDecimal priceChangePercentage7d,
    BigDecimal priceChangePercentage14d,
    BigDecimal priceChangePercentage30d,
    BigDecimal priceChangePercentage60d,
    BigDecimal priceChangePercentage200d,
    BigDecimal priceChangePercentage1y,

    BigDecimal circulatingSupply,
    BigDecimal totalSupply,
    BigDecimal maxSupply,

    String ath,
    BigDecimal athChangePercentage,
    String athDate,
    String atl,
    BigDecimal atlChangePercentage,
    String atlDate,

    List<Double> sparkline7d,
    String updatedAt
) {}

File: app/xchange_app/src/types/coin-detail-response.ts
export interface CoinDetailResponse {
  id: string
  name: string
  symbol: string
  imageUrl: string
  description: string
  hashingAlgorithm: string
  genesisDate: string
  marketCapRank: number
  watchlistPortfolioUsers: number
  sentimentVotesUpPercentage: number
  sentimentVotesDownPercentage: number

  price: string
  high24h: string
  low24h: string
  priceChange24h: string
  marketCap: string
  totalVolume: string
  fullyDilutedValuation: string

  priceChangePercentage1h: number
  priceChangePercentage24h: number
  priceChangePercentage7d: number
  priceChangePercentage14d: number
  priceChangePercentage30d: number
  priceChangePercentage60d: number
  priceChangePercentage200d: number
  priceChangePercentage1y: number

  circulatingSupply: number
  totalSupply: number
  maxSupply: number

  ath: string
  athChangePercentage: number
  athDate: string
  atl: string
  atlChangePercentage: number
  atlDate: string

  sparkline7d: number[]
  updatedAt: string
}
